app/src/main/java/io/neurolab/activities/MeditationActivity.java

```diff
@@ -126,9 +128,14 @@ public void onStopTrackingTouch(SeekBar seekBar) {
                 });
     }
 
+    @Override
+    public boolean onSupportNavigateUp(){
+        onBackPressed();
+        return true;
+    }
+
     @Override
     public void onBackPressed() {
-        super.onBackPressed();
         finish();
```

Comment:
update this as well, you should not need to finish activity on back press. It is the default nature of any activity